pkgs/dash_analytics/lib/src/analytics.dart

```diff
@@ -127,7 +127,7 @@ abstract class Analytics {
   /// API to send events to Google Analytics to track usage
   Future<Response>? sendEvent({
     required DashEvent eventName,
-    required Map<String, Object?> eventData,
+    Map<String, Object?> eventData,
```

Comment:
Since we're removing `required` here, wouldn't that change the signature to `Map<String, Object?>?`? And / or need to assign a const default to `eventData`? I'm not sure why the analyzer isn't complaining here (or what I'm missing).

Reply:
Perhaps no default value is necessary if the method is abstract.

Reply:
It isn't actually going to be nullable, it will just default to an empty map in the implementation.

What would be best practice here, should I also include the default value in the abstract?

```dart
  // OLD
  /// API to send events to Google Analytics to track usage
  Future<Response>? sendEvent({
    required DashEvent eventName,
    Map<String, Object?> eventData,
  });

  // NEW
  /// API to send events to Google Analytics to track usage
  Future<Response>? sendEvent({
    required DashEvent eventName,
    Map<String, Object?> eventData = const <String, Object?>{},
  });
```

Both approaches work in the example file

Reply:
I don't know that I have a strong opinion here; perhaps including it in the abstract method would help server as documentation.

Also - I missed this in the initial review - but we should remove the `always_specify_types` types line from the analysis_options file (that lint generally goes counter to the dart style guide).

With that change, the above method sig can change to: `Map<String, Object?> eventData = const {},`.

Reply:
Ah makes sense, i'll remove that lint rule and update the abstract method to include the default value as well